src/typing/flow_js.ml

```diff
@@ -3066,20 +3066,14 @@ let rec __flow cx (l, u) trace =
     (*******************************************)
 
     | (FunT _,
-       MethodT(
-         reason_op,
-         (_, "apply"),
-         ({params_tlist=[o2;tinsArr2]; _} as funtype)
-       ))
-      -> (* TODO: closure *)
-
-      let reason = replace_reason "element of arguments" reason_op in
-      let elem = mk_tvar cx reason in
-
-      rec_flow cx trace (tinsArr2, ArrT(reason,elem,[]));
```

Comment:
i was thinking that doing `rec_flow cx trace (tinsArr2, ArrRestT(reason, 0, elem)` here would mean that `elem` would wait for `tinsArr2` to be resolved and let you access the `ts` you need from the `ArrT`. but I think you'd still need another type like `ApplyT` to build the `ts @ [RestT t]` param list, so I'm not sure how my suggestion would work either... and `ApplyT` is already waiting for it to be resolved so using `ArrRestT` would be useless.